src/Arena.js

```diff
@@ -1,554 +1,555 @@
+  endTurn,
+  skillBuff,
+  skill,
+  setSkill,
+  opponentDisabled,
+  poisonDmg,
+  bleedDmg,
+} from "./actions";
+import {
+  diceRoll,
+  sleep,
+  applyPoisonStyle,
+  oppGetHurt,
+  dodgeShake,
+  oppAttackAnimation,
+  closeDropup,
+  openDropup,
+  bleedTickAnimation,
+  displayDmgNumber,
+  animateBuff
+} from "./data/functions";
+import ClickNHold from "react-click-n-hold";
+import Gladiator from "./models/Gladiator";
+import Opponent from "./models/Opponent";
+
+// PNG assets
+import riposteImage from './assets/images/animations/riposte.gif';
+import { OpponentType } from "./data/opponentData";
 
-let stack = 0
 
 export default function Arena() {
-  
   const dispatch = useDispatch();
-  const battleState = useSelector(state => state.BattleState)
-  const myGladiator = useSelector(state => state.ChosenGladiator.chosen)
-  const opponent = useSelector(state => state.ChosenOpponent.chosenOpp)
-  const opponentArray = useSelector(state => state.OpponentReducer)
-  const allSkills = useSelector(state => state.SkillsReducer)
-  let mySkill = useSelector(state => state.SelectedSkill.mySkill)
-  let skillArray = [];
-  let activeBuffs = battleState.activeBuffs;
+  const {
+    battleState,
+    myGladiator,
+    opponent,
+    opponentArray,
+    allSkills,
+    mySkill,
+    activeBuffs,
+  } = useSelector((state) => ({
+    battleState: state.BattleState,
+    myGladiator: state.ChosenGladiator.chosen,
+    opponent: state.ChosenOpponent.chosenOpp,
+    opponentArray: state.OpponentReducer,
+    allSkills: state.SkillsReducer,
+    mySkill: state.SelectedSkill.mySkill,
+    activeBuffs: state.BattleState.activeBuffs,
+  }));
 
   // Setting skills for use in combat. skillArray will contain all usable skills by gladiator at its current level
-  allSkills.map((skill) => {
-    if (skill.gladiator === myGladiator.name && myGladiator.level >= skill.lvlUnlock && !skill.passive){
-      skillArray.push(skill)
-    }
-    return skillArray
-  });
-
-/*---------------------STAT DECLARATIONS-------------------------*/
-    let opponentHP = opponent.hp + battleState.opponentHealth
-    let currentHP = myGladiator.hp + battleState.playerHealth
-    let myHit
-    let oppHit
-    let atkRoll
-    let critRoll
-    let passiveEffect
-    let bleedTick
-    let myAC = myGladiator.ac
-    let critChance = myGladiator.critChance
-    let dmgBonus = myGladiator.dmgBonus
-    let toHitBonus = myGladiator.toHitBonus
-    let oppToHitBonus = opponent.toHitBonus
-    let blockValue = myGladiator.blockValue
-
-//Applies Dex and STR bonuses to hit and damage
-    if (myGladiator.dex > 15){
-      toHitBonus += (myGladiator.dex - 15)
-    }
+  const skillArray = allSkills.filter(
+    (skill) => skill.gladiator === myGladiator.name && myGladiator.level >= skill.lvlUnlock && !skill.passive
+  );
 
-    if (myGladiator.str > 15){
-      dmgBonus += (myGladiator.str - 15)
-    }
+  const [gladiator] = useState(new Gladiator(myGladiator, activeBuffs, dispatch));
```

Comment:
Here's where we can use React state to hold a reference to the gladiator for the duration of the fight in the Arena.

This is usually a complicated thing to do to mix redux state and react state where they may overlap - in this case here's the high level idea:

There's a "chosen gladiator" with a base set of stats, abilities etc. and he's in the redux store.
There's another "mutable" gladiator who has calculated hp, other stats, that change during the course of the battle. That's handy in React state since it initializes with the component here and then goes away once the component unmounts.